src/core/server/saved_objects/migrations/types.ts

```diff
@@ -65,6 +65,10 @@ export interface SavedObjectMigrationContext {
    * The version in which this object type is being converted to a multi-namespace type
    */
   readonly convertToMultiNamespaceTypeVersion?: string;
+  /**
+   * Whether this is a single-namespace type or not
+   */
+  readonly isSingleNamespaceType: boolean;
```

Comment:
I was under the impression that as of 8.0.0 we wouldn't have any `single-namespace` SO types anymore. If I'm not correct, then in what situations would we use these types? config maybe?

Reply:
Yeah, right now `config` is the only type that should only ever be single-namespace. No other object types have been identified as of yet that will only ever need to exist in a single space.

If `config` turns out to be the only one, we could realistically convert that too, but we would have to change how its ID is set. Then we could get rid of `namespaceType: 'single'` completely.